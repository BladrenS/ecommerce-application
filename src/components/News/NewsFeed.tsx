import axios from 'axios';
import { useEffect, useState } from 'react';

import styles from './styles.module.scss';

type Article = {
  title: string;
  description: string;
  url: string;
  image: string;
};

export const NewsFeed = () => {
  const [articles, setArticles] = useState<Article[]>([]);

  useEffect(() => {
    const fetchNews = async () => {
      try {
        const response = await axios.get('https://gnews.io/api/v4/search', {
          params: {
            q: 'motorcycles OR yamaha OR kawasaki OR motoGP',
            lang: 'en',
            country: 'any',
            max: 25,
            apikey: 'af607d434e29d1cf96d13223350b456b',
          },
        });
        setArticles(response.data.articles);
      } catch (error) {
        console.error('Ошибка при загрузке новостей:', error);
      }
    };

    fetchNews();
  }, []);

  return (
    <div className={styles.news}>
      <ul>
        {articles.map((article, index) => (
          <li key={index} style={{ marginBottom: '20px' }}>
            <a href={article.url} target="_blank" rel="noopener noreferrer">
              <h3>{article.title}</h3>
            </a>
            <p>{article.description}</p>
            {article.image && (
              <img
                src={article.image}
                alt={article.title}
                style={{ width: '100%', maxWidth: '400px', borderRadius: '8px' }}
              />
            )}
          </li>
        ))}
      </ul>
    </div>
  );
};
/*type Article = {
  title: string;
  description: string;
  url: string;
  urlToImage: string;
};

export const NewsFeed = () => {
  const [articles, setArticles] = useState<Article[]>([]);

  useEffect(() => {
    const fetchNews = async () => {
      try {
        const response = await axios.get('https://newsapi.org/v2/everything', {
          params: {
            q: 'yamaha OR kawasaki OR motoGP',
            sortBy: 'popularity',
            language: 'en',
            apiKey: '71ab631bae824a6588800a679f9c5e4e',
            pageSize: 15,
          },
        });
        setArticles(response.data.articles);
      } catch (error) {
        console.error('Ошибка при загрузке новостей:', error);
      }
    };

    fetchNews();
  }, []);

  return (
    <div className={styles.news}>
      <ul>
        {articles.map((article, index) => (
          <li key={index} style={{ marginBottom: '20px' }}>
            <a href={article.url} target="_blank" rel="noopener noreferrer">
              <h3>{article.title}</h3>
            </a>
            <p>{article.description}</p>
            {article.urlToImage && (
              <img
                src={article.urlToImage}
                alt={article.title}
                style={{ width: '100%', maxWidth: '400px', borderRadius: '8px' }}
              />
            )}
          </li>
        ))}
      </ul>
    </div>
  );
};*/
