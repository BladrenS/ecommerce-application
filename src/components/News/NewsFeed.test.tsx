import { render, waitFor } from '@testing-library/react';
import axios from 'axios';

import { NewsFeed } from './NewsFeed';

jest.mock('axios');

const mockedAxiosGet = jest.mocked(axios.get, { shallow: true });

describe('NewsFeed', () => {
  it('renders loading and then news items', async () => {
    mockedAxiosGet.mockResolvedValueOnce({
      data: {
        articles: [
          {
            title: 'Test Title',
            description: 'Test Description',
            url: 'https://example.com',
            urlToImage: 'https://example.com/image.jpg',
          },
        ],
      },
    });

    const { getByText, container } = render(<NewsFeed />);

    expect(container.querySelector('div')).toBeInTheDocument();

    await waitFor(() => expect(getByText('Test Title')).toBeInTheDocument());

    expect(container).toMatchSnapshot();
  });
});
