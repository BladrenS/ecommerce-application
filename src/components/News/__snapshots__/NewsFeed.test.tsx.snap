// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsFeed renders loading and then news items 1`] = `
<div>
  <div
    class="news-container"
  >
    <div
      class="news-inner"
    >
      <a
        class="news-item"
        href="https://example.com"
        style="background-size: cover;"
        target="_blank"
      >
        <h3
          class="news-item-header"
        >
          Test Title
        </h3>
        <p
          class="news-item-text"
        >
          Test Description
        </p>
      </a>
    </div>
  </div>
</div>
`;
